libsql/src/replication/connection.rs

```diff
@@ -191,6 +191,10 @@ impl RemoteConnection {
                 .into();
         }
 
+        if let Some(replicator) = writer.replicator() {
+            replicator.sync_oneshot().await?;
+        }
```

Comment:
To clarify - this is the important bit that gives us read-before-wrote, right? Combined with the fact that writer.replicator is only engaged if we passed `read_before_write == true` before.

Reply:
Also, @LucioFranco, `res` returned from `execute_program` contains the latest frame number after the program got executed, right? So perhaps it makes sense to make `sync_oneshot` take a `frame_no` parameter, and sync just to this specific frame, instead of waiting until we catch up with the primary (so instead of returning from `sync_oneshot` when `replica_index >= primary_index`, we'd return when `replica_index >= that_index_we_just_received_in_the_response_above`)? If I understand correctly, it could make read-your-writes more leightweight, because we'd only wait for our write, not potentially somebody else's.

Reply:
Sure, lets do this in a follow up 👍